vsutil/__init__.py

```diff
@@ -275,3 +281,31 @@ def _resolve_enum(enum: Type[E], value: Any, var_name: str, module: Optional[str
         return enum(value)
     except ValueError:
         raise ValueError(f'{var_name} must be in {_readable_enums(enum, module)}.') from None
+
+
+def is_variable(clip: vs.VideoNode, /, *, format: bool = False, resolution: bool = False) -> bool:
+    """
+    Returns True if at least one of the specified clip's attributes is variable.
+    It is an error to use this without specifying which attribute to check.
+    """
+    if not format and not resolution:
+        raise ValueError('At least one attribute must be specified as `True`.')
+
+    if format:
+        if clip.format is None:
+            return True
+    if resolution:
+        if 0 in (clip.width, clip.height):
+            return True
+
+    return False
+
+
+def _disallow_variable_format(clip: vs.VideoNode, /) -> None:
+    if is_variable(clip, format=True):
+        raise ValueError('Variable-format clips not supported.')
+
+
+def _disallow_variable_resolution(clip: vs.VideoNode, /) -> None:
+    if is_variable(clip, resolution=True):
+        raise ValueError('Variable-resolution clips not supported.')
```

Comment:
I suggest turning these into decorators. Makes the intent clear and currently they are all used as the first line of the function anyway.

Reply:
ooooh, good idea :+1: , will implement later.

Reply:
Implemented in https://github.com/Irrational-Encoding-Wizardry/vsutil/pull/37/commits/03f90fce8bd4d8235dcf913c4d49a52a3a00699e, not sure if this is the correct way to do this since I've never created a decorator before, nevermind a properly type hinted one. Maybe @heicrd can weigh in on the type hinting. mypy seems to be happy with this, using `typing.Callable` was a mess as the function signature is not guaranteed.

Reply:
Oh, nope this is actually broken, welp.

Reply:
I'm dumb, here, fixed (I think) https://github.com/Irrational-Encoding-Wizardry/vsutil/pull/37/commits/95beb7fad0bd680597248ddbaa67b97a05fcba21

I still need to add tests for this but I've tested it locally and it works as expected, just not sure this is the correct way of doing this.